docs/script/generatePropsTables.mjs

```diff
@@ -0,0 +1,158 @@
+  // Parse a file for docgen info
+  const rawData = docgen.parse(pathToFile)[0];
+
+  if (!rawData) return {};
+
+  /**
+   * need to treat special characters
+   * 1) "|", "<", ">", "`" => replace with character code, "&$<unicode>;"
+   * 2) "' + '" => replace with space
+   * 3) "\n" => replace with space
+   */
+  const tobeEncoded = new RegExp(/[|<>`]|'\s\+\s'|\\n/g);
+  const getEncoded = (match) =>
+    match.match(/[|<>`]/) ? `&#${match.charCodeAt()};` : ' ';
+  const enCodedData = JSON.stringify(rawData).replaceAll(
+    tobeEncoded,
+    getEncoded
+  );
+
+  return JSON.parse(enCodedData);
+};
+
+const getWantedCategories = (displayName) => [
+  `${displayName}Props`,
+  'BaseComponentProps',
+  'BaseStyleProps',
+  'RefAttributes',
+];
+
+const categorizeProps = (props, wantedCategories) =>
+  Object.entries(props).reduce((acc, [key, val]) => {
+    const category = val.declarations[0].name;
+    const isCategoryToUse = [...wantedCategories, 'TypeLiteral'].includes(
+      category
+    );
+    const categoryKey =
+      category === 'TypeLiteral' ? 'BaseComponentProps' : category;
+    return {
+      ...acc,
+      ...(isCategoryToUse && {
+        [categoryKey]: {
+          ...acc[categoryKey],
+          [key]: val,
+        },
+      }),
+    };
+  }, {});
+
+const createPropsTable = (props) => {
+  return {
+    headers: ['name', 'type', 'default', 'description'],
+    rows: props
+      ? Object.values(props).map(
+          ({ name, type, defaultValue, description }) => ({
+            name: name.replaceAll(/[{}]/g, '\\$&'),
+            type: type.name.replaceAll(/[{}]/g, '\\$&'),
+            default: defaultValue ? defaultValue.value.toString() : '-',
+            description,
+          })
+        )
+      : [{ name: '-', type: '-', default: '-', description: '-' }],
+  };
+};
+
+const createPropsTableExpander = (data, wantedCategories) =>
+  wantedCategories.map((category) => ({
+    ExpanderItem: {
+      title: category,
+      value: category,
+      children: [
+        {
+          table: createPropsTable(data[category]),
+        },
+      ],
+    },
+  }));
+
+json2md.converters.ExpanderItem = ({ title, value, children }, json2md) => `
+<ExpanderItem title="${title}" value="${value}">
+    <ReactMarkdown
+      remarkPlugins={[remarkGfm]}
+      children={\`
+${json2md(children)}
+      \`}
+    />
+  </ExpanderItem>
+`;
+
+json2md.converters.Expander = ({ ExpanderItems, displayName }, json2md) => `
+<Expander type="multiple" defaultValue={['${displayName}Props']}>
+  ${json2md([...ExpanderItems.map((item) => ({ ...item }))])}
+</Expander>
+`;
+
+json2md.converters.plainText = (text, json2md) => text;
+
+const getOutput = (displayName, propsTables) =>
+  json2md([
+    {
+      plainText: `
+import { Expander, ExpanderItem } from '@aws-amplify/ui-react';
+import ReactMarkdown from 'react-markdown';
+import remarkGfm from 'remark-gfm';
+`,
+    },
+    { h2: `${displayName} Props` },
```

Comment:
nit: Maybe change this to just 'Props'? I think repeating the component name is not necessary for the heading here (and so it will also just say "Props" in the jump nav on the right).

Reply:
Removed